src/tests/test_parser.rs

```diff
@@ -508,6 +508,28 @@ fn list_type_annotations_nested_with_err() -> TestResult {
     fail_test(input, expected)
 }
 
+#[test]
+fn list_type_annotations_nested_with_unknown_type() -> TestResult {
+    let input =
+        "def test [list: list<list<str>>] { $list | length }; test [ ['nushell'], ['nunu'] ]";
```

Comment:
Took me a second to realize that `str` is not `string`. Damn you Rust and Python brain. :D